File: bcs-ui/backend/container_service/projects/cmdb.py
# -*- coding: utf-8 -*-
"""
Tencent is pleased to support the open source community by making 蓝鲸智云PaaS平台社区版 (BlueKing PaaS Community
Edition) available.
Copyright (C) 2017-2021 THL A29 Limited, a Tencent company. All rights reserved.
Licensed under the MIT License (the "License"); you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://opensource.org/licenses/MIT

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
"""
from typing import List

from backend.components import cc


def list_biz_maintainers(biz_id: int) -> List[str]:
    """查询业务的运维角色"""
    return cc.get_app_maintainers("admin", biz_id)


def is_biz_maintainer(biz_id: int, username: str) -> bool:
    """判断用户是否为业务的运维角色"""
    maintainers = list_biz_maintainers(biz_id)
    return username in maintainers

File: bcs-services/bcs-platform-manager/pkg/api/cmdb/cmdb.go
/*
 * Tencent is pleased to support the open source community by making Blueking Container Service available.
 * Copyright (C) 2019 THL A29 Limited, a Tencent company. All rights reserved.
 * Licensed under the MIT License (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 * http://opensource.org/licenses/MIT
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Package cmdb cmdb operate
package cmdb

import (
	"context"

	actions "github.com/Tencent/bk-bcs/bcs-services/bcs-platform-manager/pkg/actions/cmdb"
	"github.com/Tencent/bk-bcs/bcs-services/bcs-platform-manager/pkg/component/cmdb"
	"github.com/Tencent/bk-bcs/bcs-services/bcs-platform-manager/pkg/types"
)

// DeleteAllByBkBizIDAndBkClusterID 清理cmdb容器数据
// @Summary 清理cmdb容器数据
// @Tags    Logs
// @Produce json
// @Success 200 {bool} bool
// @Router  /cmdb/delete_all [put]
func DeleteAllByBkBizIDAndBkClusterID(ctx context.Context,
	req *types.DeleteAllByBkBizIDAndBkClusterIDReq) (*bool, error) {
	result, err := actions.NewCmdbAction().DeleteAllByBkBizIDAndBkClusterID(ctx, req)
	if err != nil {
		return nil, err
	}

	return &result, nil
}

// GetBusiness 获取business列表
// @Summary 获取business列表
// @Tags    Logs
// @Produce json
// @Success 200 {bool} bool
// @Router  /cmdb/business [post]
func GetBusiness(ctx context.Context,
	req *types.DeleteAllByBkBizIDAndBkClusterIDReq) (*[]cmdb.GetBusinessRespDataInfo, error) {
	result, err := actions.NewCmdbAction().GetBusiness(ctx)
	if err != nil {
		return nil, err
	}

	return result, nil
}
